### PR TITLE
Fix Hot start file Checkpoints in Opt Manager

### PR DESCRIPTION
**Pull request by sean-engelstad**
* Bug in optimization manager where it is not copying the hot start file checkpoints
* If statement to not attempt to copy hot start files when it's not provided

### CODE REVIEW

File: funtofem/optimization/optimization_manager.py [hunk 1]
```diff
@@ -192,11 +192,12 @@ def _gatekeeper(self, x_dict):
                     self.comm, func_file, full_precision=False, optim=True
                 )
                 # copy the hotstart file to the checkpoints folder
-                src = self.hot_start_file
-                dest = os.path.join(
-                    self._checkpoints_folder, f"hot_start{self._iteration}.hst"
-                )
-                shutil.copy(src, dest)
+                if self.hot_start_file:
```
Comment (bburke38) [inline funtofem/optimization/optimization_manager.py:195]:
```suggestion
                if self.hot_start_file is not None:
```

Comment (bburke38) [reply funtofem/optimization/optimization_manager.py:195]: Would it be better to form it like this?

Comment (sean-engelstad) [reply funtofem/optimization/optimization_manager.py]: It's equivalent but more readable in the second way.

Comment (sean-engelstad) [reply funtofem/optimization/optimization_manager.py]: I went ahead and made your change using your comment.

**Pull request merged**